tests/integration/test_dbapi_integration.py

```diff
@@ -123,22 +124,114 @@ def test_string_query_param(trino_connection):
     assert rows[0][0] == "six'"
 
 
+def test_decimal_query_param(trino_connection):
+    cur = trino_connection.cursor()
+
+    cur.execute("SELECT ?", params=(Decimal('0.142857'),))
```

Comment:
we should test with:

- null
- decimal(38, 0)
- largest supported Trino decimal(38, 38)
- smallest supported Trino decimal(38, 38)

Also are we losing the scale and precision information? We should test whether the scale and precision of the input is being preserved.

Reply:
This is now added as `test_null_decimal`, `test_biggest_decimal`, `test_smallest_decimal` and `test_highest_precision_decimal`.